MatrixKit/Controllers/MXKRoomAttachmentsViewController.m

```diff
@@ -0,0 +1,666 @@
+{
+    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    
+    // Store index of the current displayed attachment, to restore it after refreshing
+    currentAttachmentIndex = _attachmentsCollection.contentOffset.x / [[UIScreen mainScreen] bounds].size.width;
+    
+    // Show temporarily the navigation bar (required in case of splitviewcontroller use)
+    self.navigationController.navigationBarHidden = NO;
+    navigationBarDisplayTimer = [NSTimer scheduledTimerWithTimeInterval:3 target:self selector:@selector(hideNavigationBar:) userInfo:self repeats:NO];
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(coordinator.transitionDuration * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        
+        // Cell width will be updated, force collection refresh to take into account changes.
+        [_attachmentsCollection reloadData];
+        
+        // Refresh the current attachment display
+        [self refreshAttachmentCollectionContentOffset];
+        
+    });
+}
+
+// The 2 following methods are deprecated since iOS 8
+- (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
+{
+    [super willRotateToInterfaceOrientation:toInterfaceOrientation duration:duration];
+    
+    // Store index of the current displayed attachment, to restore it after refreshing
+    currentAttachmentIndex = _attachmentsCollection.contentOffset.x / [[UIScreen mainScreen] bounds].size.width;
+    
+    // Show temporarily the navigation bar (required in case of splitviewcontroller use)
+    self.navigationController.navigationBarHidden = NO;
+    navigationBarDisplayTimer = [NSTimer scheduledTimerWithTimeInterval:3 target:self selector:@selector(hideNavigationBar:) userInfo:self repeats:NO];
+}
+- (void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceOrientation
+{
+    [super didRotateFromInterfaceOrientation:fromInterfaceOrientation];
+    
+    // Cell width will be updated, force collection refresh to take into account changes.
+    dispatch_async(dispatch_get_main_queue(), ^{
+        
+        [_attachmentsCollection reloadData];
+        
+        // Refresh the current attachment display
+        [self refreshAttachmentCollectionContentOffset];
+        
+    });
+}
+
+#pragma mark - Override MXKViewController
+
+- (void)destroy
+{
+    [super destroy];
+}
+
+#pragma mark - Public API
+
+- (void)displayAttachments:(NSArray*)attachmentArray focusOn:(NSString*)eventId
+{
+    attachments = [NSMutableArray arrayWithArray:attachmentArray];
+    
+    currentAttachmentIndex = 0;
+    
+    for (NSUInteger index = 0; index < attachments.count; index++)
+    {
+        MXKAttachment *attachment = attachments[index];
+        
+        if ([attachment.event.eventId isEqualToString:eventId])
+        {
+            currentAttachmentIndex = index;
+        }
+    }
+    
```

Comment:
1 - currentAttachmentIndex should be updated if the item is found.
2 - a break or a return + reloadData could be done in the if block to same time